Start working through warnings and notes when running devtools::check

nikosbosse opened the pull request:
When I built and checked the package locally a lot of errors, warnings and notes came up. 

I started working through some of them, but it was taking me more time than I expected. I think it would be good to work through the problems one by one until devtools::check passes. 

This PR mostly deals with some NAMESPACE issues and imports packages explicitly, but I haven't finished working on it yet.

elray1 commented on DESCRIPTION:
```diff
@@ -28,7 +28,31 @@ Description:
+    httr,
+    stringr,
+    yaml,
+    covidData
```
r.e. this dependency on covidData -- did you notice functions in the package that needed this other than `preprocess_jhu`?  Our hope is that a typical user of covidHubUtils would be able to use the package without having covidData installed; I think most other functions don't rely on its presence.  Did you notice other functions that relied on covidData, or were you getting errors if covidData was not listed in the imports?

`preprocess_jhu` is only used by us to output truth files to the hub repo, and we only put it into this package to give it a place where it could be formally and routinely unit tested. But if its presence is causing problems, my preference would be to move it somewhere else or figure out the right settings to make covidData optional-but-not-required.  And if there are other functions depending on covidData here, similar comments apply.

nikosbosse replied:
There is also `preprocess_hospitalization` that uses it. 

I think I got a warning that funcitons were called using `covidData::` without it being an import - that's why I added it to imports.

elray1 reviewed (commented):
All the changes look good to me, except that I have a strong preference not to introduce a dependency on covidData in this package.  More detailed comment in `DESCRIPTION` file.

nikosbosse commented:
ah, I simply wasn't aware of the workflow. I got an warning as there is a function call using :: involving covidData, so I assumed it should be added to imports. 

I updated the PR and added covidData to 'Suggests'. Not entirely sure what the best way to deal with this is.

elray1 commented on R/utils.R:
```diff
@@ -0,0 +1,16 @@
+globalVariables(
+  c(".", "County", "Name", "Prediction", "Interval", "State", "StateFIPS", 
+    "V1", "V2", "abbreviation", "alpha", "cases", "cases_deaths", "county", 
+    "countyFIPS", "cum", "deaths", "designation", "endpoint_type", 
+    "exists_interval_score_0", "fips", "forecast_date", "full_location_name", 
+    "geo_type", "head", "horizon", "inc", "inc_cases", "inc_deaths", 
+    "interval_score_0", "location", "location_name", "lower", "model", 
+    "model.x", "model_abbr", "n_interval_scores", "notes", "overprediction_0", 
+    "point", "quantile", "sharpness_0", "state", "state_abbreviation", 
+    "tail", "target", "target_end_date", "target_variable", 
+    "team_model_designation", "temporal_resolution", "timezero", "true_value", 
+    "truth_forecast", "truth_model", "type", "underprediction_0", 
+    "unit", "upper", "value", "value.x", "value.y", 
+    "County", "Name", "Prediction", "Interval"
+  )
+)
```
It seems like these are column names in various data frames that are referenced within tidyverse operations like `dplyr::mutate`, and not actually global variables defined by the package.  For example, the variables like `overprediction_0`, `sharpness_0`, and `underprediction_0` are referenced at https://github.com/reichlab/covidHubUtils/blob/b4e5fa21d878c8c95d21745191346a499e6e8495/R/score_forecasts.R#L157.  It seems like declaring them as `globalVariables` is probably not the right thing to do here.  Should we delete this?  If you were prompted to create this by something like `devtools::check()` or similar, does that suggest we should make some other change at the point these variables are referenced to satisfy the check?

elray1 replied:
I found this thread which suggests all those mutate statements etc should use `.data$underprediction_0` and similar, after importing `.data` from `rlang`: https://community.rstudio.com/t/when-programming-with-dplyr-what-is-the-correct-way-to-avoid-undefined-global-variables/55946

Maybe we could delete this `globalVariables` call for now and create a separate issue to clean those up?

nikosbosse replied:
hm I always defined them as globalVariables to get rid of the note (and also have seen that elsewhere in other packages), but I admit I don't _really_ _really_ understand what is going on underneath the hood. So happy to do it either way. Would you like me to make a change? Or do you just want to change it yourself?

nikosbosse replied:
Maybe @seabbs has smart thoughts?

elray1 replied:
I think this is not a big deal, so I'd be fine with merging this in for now and filing a separate issue for later to go back and revise the calls to mutate

nikosbosse replied:
:+1: sounds good to me

Serena-Wang commented:
Hi @nikosbosse, I've updated the GitHub action workflow file to skip packages in `suggests` section when it's installing all dependencies. Please pull it from the main branch. It should fix the unit test.

nikosbosse commented:
Hi @Serena-Wang, I updated our fork with the change. My problem, however, is still that the unit tests depend on the covidData package. Not sure that solves that? Maybe it would be a good idea, as Evan suggested, to move the unit tests that depend on covidData somewhere else at some point.

elray1 commented:
> Hi @Serena-Wang, I updated our fork with the change. My problem, however, is still that the unit tests depend on the covidData package. Not sure that solves that? Maybe it would be a good idea, as Evan suggested, to move the unit tests that depend on covidData somewhere else at some point.

We think that if this change were part of the PR, it should make the github action that runs the unit tests pass, as those tests that depend on covidData are not run as part of that process.  This would just be a reassuring step before going ahead with the merge :)

nikosbosse commented:
gotcha :)
